tests/execution.rs

```diff
@@ -3435,6 +3453,69 @@ fn test_invalid_exit_or_return() {
     }
 }
 
+#[test]
+fn callx_unsupported_instruction_and_exceeded_max_instructions() {
+    test_interpreter_and_jit_asm!(
+        "
+        sub32 r7, r1
+        sub64 r5, 8
+        sub64 r7, 0
+        callx r5
+        callx r5
+        return
+        ",
+        [],
+        TestContextObject::new(4),
+        ProgramResult::Err(EbpfError::UnsupportedInstruction),
+    );
+
+    let prog = &[
+        0x1c, 0x17, 0x17, 0x95, 0x61, 0x61, 0x61, 0x61, // sub32
+        0x17, 0x55, 0x54, 0xff, 0x08, 0x00, 0x00, 0x00, // sub64
+        0x17, 0x17, 0x17, 0xff, 0x00, 0x00, 0x00, 0x00, // sub64
+        0x8d, 0x55, 0x54, 0xff, 0x09, 0x00, 0x00, 0x00, // callx
+        0x8d, 0x55, 0x54, 0x23, 0x08, 0x00, 0x11, 0x4e, // callx
+    ];
+
+    let loader = Arc::new(BuiltinProgram::new_loader(
+        Config::default(),
+        FunctionRegistry::default(),
+    ));
+    let mut executable = Executable::<TestContextObject>::from_text_bytes(
+        prog,
+        loader,
+        SBPFVersion::V2,
+        FunctionRegistry::default(),
+    )
+    .unwrap();
+
+    test_interpreter_and_jit!(
```

Comment:
Isn't this the same as above?

Reply:
https://github.com/solana-labs/rbpf/blob/8d36530b7071060e2837ebb26f25590db6816048/tests/execution.rs#L51-L53

The `test_interpreter_and_jit_asm` macro defines a 1024 CU budget if the error is not `ExceededMaxInstructions` (see above) , so the first example runs without any budget constraint. In the second, I override the setting for that not to happen.